crates/biome_js_analyze/src/lint/nursery/no_useless_escape_in_string.rs

```diff
@@ -0,0 +1,180 @@
+use biome_analyze::{context::RuleContext, declare_lint_rule, Ast, FixKind, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_diagnostics::Severity;
+use biome_js_syntax::{
+    AnyJsTemplateElement, JsLiteralMemberName, JsStringLiteralExpression, JsSyntaxKind,
+    JsSyntaxToken, JsTemplateExpression,
+};
+use biome_rowan::{declare_node_union, BatchMutationExt, TextRange};
+
+use crate::JsRuleAction;
+
+declare_lint_rule! {
+    /// Disallow unnecessary escapes in string literals.
+    ///
+    /// Escaping non-special characters in string literals doesn't have any effect.
+    /// Hence, they may confuse a reader.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// const s = "\a";
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const o = {
+    ///     "\a": 0,
+    /// };
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const s = `${0}\a`;
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// const s = "\n";
+    /// ```
+    ///
+    /// Tagged string template are ignored:
+    ///
+    /// ```js
+    /// const s = tagged`\a`;
+    /// ```
+    ///
+    /// JSX strings are ignored:
+    ///
+    /// ```jsx
+    /// <div attr="str\a"/>;
+    /// ```
+    ///
+    pub NoUselessEscapeInString {
+        version: "next",
+        name: "noUselessEscapeInString",
+        language: "js",
+        recommended: true,
+        severity: Severity::Error,
```

Comment:
In a recent PR I demoted the severity of all the rules that have "useless" in their name to warning or information. The reason is because, as the name says, these are rules that trigger some useless code, which isn't an error.

We should demote the severity or changing the rule name

Reply:
I chose this severity level because it is the one set for `noUselessEscapeInRegex`. I can change the two severity levels.

Reply:
Done